forecasting/src/autogluon/forecasting/models/presets.py

```diff
@@ -0,0 +1,198 @@
+                ),
+            },
+            "SimpleFeedForward": {
+                "context_length": ag.Int(
+                    min(prediction_length, max(10, 2 * prediction_length), 250),
+                    max(min(500, 12 * prediction_length), prediction_length),
+                    default=prediction_length,
+                ),
+            },
+        },
+    }
+    return default_model_hps[key]
+
+
+def get_preset_models(
+    freq: str,
+    prediction_length: int,
+    path: str,
+    eval_metric: str,
+    hyperparameters: Union[str, Dict],
+    hyperparameter_tune: bool,
+    invalid_model_names: List[str],
+    **kwargs,
+):
+    """
+    Create a list of models according to hyperparameters. If hyperparamaters=None,
+    will create models according to presets.
+    """
+    models = []
+    if isinstance(hyperparameters, str):
+        hyperparameters = copy.deepcopy(
+            get_default_hps(hyperparameters, prediction_length)
+        )
+    else:
+        hp_str = "default" if not hyperparameter_tune else "default_hpo"
+        default_hps = copy.deepcopy(get_default_hps(hp_str, prediction_length))
+
+        if hyperparameters is not None:
+            # filter only default_hps for models with hyperparameters provided
+            default_hps = {
+                model: default_hps.get(model, {}) for model in hyperparameters
+            }
+            for model in hyperparameters:
+                default_hps[model].update(hyperparameters[model])
+        hyperparameters = copy.deepcopy(default_hps)
+
+    if hyperparameter_tune:
+        verify_contains_searchspace(hyperparameters)
+    else:
+        verify_no_searchspace(hyperparameters)
+
+    for model, model_hps in hyperparameters.items():
+        if isinstance(model, str):
+            if model not in MODEL_TYPES:
+                raise ValueError(f"Model {model} is not supported yet.")
+            model_type = MODEL_TYPES[model]
+        elif isinstance(model, AbstractForecastingModelFactory):
+            model_type = model
+        elif not issubclass(model, AbstractForecastingModel):
+            logger.warning(
+                f"Customized model {model} does not inherit from {AbstractForecastingModel}"
+            )
+            model_type = model
+        else:
+            logger.log(20, f"Custom Model Type Detected: {model}")
+            model_type = model
+
+        model_type_kwargs = dict(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            eval_metric=eval_metric,
+            hyperparameters=model_hps,
+            **kwargs,
+        )
+
+        # add models while preventing name collisions
+        model = model_type(**model_type_kwargs)
+        if model.name in [m.name for m in models] + invalid_model_names:
+            model = model_type(
+                name=f"{model.name}_{str(uuid.uuid4())[:6]}", **model_type_kwargs
+            )
```

Comment:
Thanks for pointing this out. I agree with points  2-3 and especially 4. The reason I went for this is since one has to keep track of state of name prefixes, or "infer" it like the code you referenced which I thought was overkill. I'll revise this.